diff --git a/drivers/mxc/ipu3/ipu_param_mem.h b/drivers/mxc/ipu3/ipu_param_mem.h
index 2ff622b5..d94a6b5 100644
--- a/drivers/mxc/ipu3/ipu_param_mem.h
+++ b/drivers/mxc/ipu3/ipu_param_mem.h
@@ -272,12 +272,12 @@ static inline void _ipu_ch_param_init(struct ipu_soc *ipu, int ch,
 	/* EBA is 8-byte aligned */
 	ipu_ch_param_set_field(&params, 1, 0, 29, addr0 >> 3);
 	ipu_ch_param_set_field(&params, 1, 29, 29, addr1 >> 3);
-	if (addr0%8)
-		dev_warn(ipu->dev,
-			 "IDMAC%d's EBA0 is not 8-byte aligned\n", ch);
-	if (addr1%8)
-		dev_warn(ipu->dev,
-			 "IDMAC%d's EBA1 is not 8-byte aligned\n", ch);
+	//if (addr0%8)
+	//	dev_warn(ipu->dev,
+	//		 "IDMAC%d's EBA0 is not 8-byte aligned\n", ch);
+	//if (addr1%8)
+	//	dev_warn(ipu->dev,
+	//		 "IDMAC%d's EBA1 is not 8-byte aligned\n", ch);
 
 	switch (pixel_fmt) {
 	case IPU_PIX_FMT_GENERIC:
