diff --git a/src/g2d/allocator.c b/src/g2d/allocator.c
index 1375342..509d6e7 100644
--- a/src/g2d/allocator.c
+++ b/src/g2d/allocator.c
@@ -52,7 +52,8 @@ GstAllocator* gst_imx_g2d_allocator_new(void)
 static gboolean gst_imx_g2d_alloc_phys_mem(GstImxPhysMemAllocator *allocator, GstImxPhysMemory *memory, gssize size)
 {
 	/* allocate cacheable physically contiguous memory block */
-	struct g2d_buf *buf = g2d_alloc(size, 1);
+	/* HYP-13887 use NOT cachable memory */
+	struct g2d_buf *buf = g2d_alloc(size, 0);
 
 	if (buf == NULL)
 	{
