From cb3c9e128c5a5209e168f38d62408d93dcf48131 Mon Sep 17 00:00:00 2001
From: Brilliantov Kirill Vladimirovich <brilliantov@byterg.ru>
Date: Wed, 25 Feb 2015 12:00:56 +0300
Subject: [PATCH] net/core/tso.c: implicit declaration of function 'tcp_hdrlen'

Signed-off-by: Brilliantov Kirill Vladimirovich <brilliantov@byterg.ru>
---
 net/core/tso.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/core/tso.c b/net/core/tso.c
index 097821d..417f30b 100644
--- a/net/core/tso.c
+++ b/net/core/tso.c
@@ -1,5 +1,6 @@
 #include <net/ip.h>
 #include <net/tso.h>
+#include <linux/tcp.h>
 
 /* Calculate expected number of TX descriptors */
 int tso_count_descs(struct sk_buff *skb)
-- 
2.1.4

