From 9da64daa24fd3f296a83979b1d8d31fd41b1acad Mon Sep 17 00:00:00 2001
From: Sascha Dierberg <dierberg@dresearch-fe.de>
Date: Mon, 27 Feb 2017 12:37:11 +0100
Subject: [PATCH] param check added to avoid encoder crash on flush HYP-15795

Signed-off-by: Sascha Dierberg <dierberg@dresearch-fe.de>
---
 src/vpu/encoder_base.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vpu/encoder_base.c b/src/vpu/encoder_base.c
index 92537a1..bf08c86 100644
--- a/src/vpu/encoder_base.c
+++ b/src/vpu/encoder_base.c
@@ -804,7 +804,7 @@ static gboolean gst_imx_vpu_encoder_flush(GstVideoEncoder *encoder)
 	ImxVpuEncReturnCodes ret;
 	GstImxVpuEncoderBase *vpu_encoder_base = GST_IMX_VPU_ENCODER_BASE(encoder);
 
-	if ((ret = imx_vpu_enc_flush(vpu_encoder_base->encoder)) != IMX_VPU_ENC_RETURN_CODE_OK)
+	if (vpu_encoder_base->encoder && ((ret = imx_vpu_enc_flush(vpu_encoder_base->encoder)) != IMX_VPU_ENC_RETURN_CODE_OK))
 		GST_ERROR_OBJECT(vpu_encoder_base, "could not flush encoder: %s", imx_vpu_enc_error_string(ret));
 
 	return ret == IMX_VPU_ENC_RETURN_CODE_OK;
-- 
2.6.6

