From fcd79ed916d098195b56ea2fba303eb6ad24cab6 Mon Sep 17 00:00:00 2001
From: Frank Bauernoeppel <frank@bauernoeppel.de>
Date: Mon, 2 Mar 2015 14:33:59 +0100
Subject: [PATCH] re-implemented CONFIG_MX6_VPU_352M

---
 arch/arm/mach-imx/clk-imx6q.c | 8 ++++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index e09b1e8..573ae98 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -691,6 +692,13 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 	/* Set pll4_audio to a value that can derive 5K-88.2KHz and 8K-96KHz */
 	clk_set_rate(clk[pll4_audio_div], 541900800);
 
+#ifdef CONFIG_MX6_VPU_352M
+   	/* see https://community.freescale.com/message/433548#433548 */
+	clk_set_rate(clk[pll2_pfd2_396m], 352000000);
+	clk_set_parent(clk[vpu_axi_sel], clk[pll2_pfd2_396m]);
+	pr_info("VPU 352M is enabled!\n");
+#endif	/* #ifdef MX6_VPU_352M */
+
 	/* Set initial power mode */
 	imx6_set_lpm(WAIT_CLOCKED);
 
-- 
1.9.1

