Index: git/mkspecs/linux-oe-g++/qmake.conf
===================================================================
--- git.orig/mkspecs/linux-oe-g++/qmake.conf	2016-12-14 17:03:17.000000000 -0600
+++ git/mkspecs/linux-oe-g++/qmake.conf	2016-12-14 17:06:23.000000000 -0600
@@ -32,6 +32,8 @@

 QMAKE_AR_LTCG      = $$(OE_QMAKE_AR_LTCG) cqs

+EGLFS_DEVICE_INTEGRATION = eglfs_viv
+
 include(../oe-device-extra.pri)

 load(device_config)
