From 02e6f94b95eb6b30d9e076de856ae603de75e0a9 Mon Sep 17 00:00:00 2001
From: Eik Binschek <binschek@dresearch-fe.de>
Date: Fri, 19 Jun 2015 16:04:55 +0200
Subject: [PATCH] Hipos: do not init wlan via udev HYP-11339

Signed-off-by: Eik Binschek <binschek@dresearch-fe.de>
---
 network.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/network.sh b/network.sh
index 3ee9271..1f365ac 100644
--- a/network.sh
+++ b/network.sh
@@ -2,6 +2,7 @@
 
 # We get two "add" events for hostap cards due to wifi0
 echo "$INTERFACE" | grep -q wifi && exit 0
+echo "$INTERFACE" | grep -q wlan && exit 0
 
 # udevd does clearenv(). Export shell PATH to children.
 export PATH
-- 
2.4.3

