From fa57d40220ca746d697c6df6807fa8eaa57378db Mon Sep 17 00:00:00 2001
From: kas User <kas@example.com>
Date: Fri, 9 Jan 2026 14:12:37 +0100
Subject: [PATCH] bootm: Authenticate image with IMX HAB

Upstream-Status: Inappropriate [product specific]
---
 boot/bootm.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/boot/bootm.c b/boot/bootm.c
index cb61485c226..053cc671713 100644
--- a/boot/bootm.c
+++ b/boot/bootm.c
@@ -778,6 +778,16 @@ int do_bootm_states(struct cmd_tbl *cmdtp, int flag, int argc,
 
 	images->state |= states;
 
+	if (IS_ENABLED(CONFIG_IMX_HAB) && states & BOOTM_STATE_START) {
+		/* validate artifact before processing it further */
+		ulong img_addr = bootm_data_addr(argc, argv);
+		if (authenticate_image(img_addr, fit_get_size((void *)img_addr))) {
+			printf("Image authentication FAILED\n");
+			return 1;
+		} else {
+			printf("Image authentication OK\n");
+		}
+	}
 	/*
 	 * Work through the states and see how far we get. We stop on
 	 * any error.
-- 
2.53.0

