From da1c1fed52367432cb8041c246d43a230309cec3 Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Thu, 24 Jun 2021 10:34:18 +0200
Subject: [PATCH] Clear DOPs again

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 libgpsd_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgpsd_core.c b/libgpsd_core.c
index 52bf8e5..e5a1339 100644
--- a/libgpsd_core.c
+++ b/libgpsd_core.c
@@ -1887,7 +1887,7 @@ void gpsd_zero_satellites( struct gps_data_t *out)
         out->skyview[sat].ss = NAN;
         out->skyview[sat].freqid = -1;
     }
-#if 0
+#if 1
     /*
      * We used to clear DOPs here, but this causes misbehavior on some
      * combined GPS/GLONASS/QZSS receivers like the Telit SL869; the
-- 
2.26.2

