From e136db9a9643f15338c3d30915089c92c15af9fa Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Mon, 9 Aug 2021 07:40:58 +0200
Subject: [PATCH] Clear DOPs again

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 gpsd/libgpsd_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gpsd/libgpsd_core.c b/gpsd/libgpsd_core.c
index ea72993..7775377 100644
--- a/gpsd/libgpsd_core.c
+++ b/gpsd/libgpsd_core.c
@@ -1967,7 +1967,7 @@ void gpsd_zero_satellites( struct gps_data_t *out)
         out->skyview[sat].ss = NAN;
         out->skyview[sat].freqid = -1;
     }
-#if 0
+#if 1
     /*
      * We used to clear DOPs here, but this causes misbehavior on some
      * combined GPS/GLONASS/QZSS receivers like the Telit SL869; the
-- 
2.26.2

