diff -Nurd git.orig/arch/arm/boot/dts/Makefile git/arch/arm/boot/dts/Makefile
--- git.orig/arch/arm/boot/dts/Makefile	2014-10-01 12:58:15.859494120 +0200
+++ git/arch/arm/boot/dts/Makefile	2014-10-01 12:59:56.544946489 +0200
@@ -119,6 +119,7 @@
 	imx6dl-nitrogen6x.dtb \
 	imx6dl-nitrogen6_max.dtb \
 	imx6dl-nitrogen6_mtp.dtb \
+	imx6dl-himx0280.dtb \
 	imx6dl-per.dtb \
 	imx6dl-sabreauto.dtb \
 	imx6dl-sabreauto-ecspi.dtb \
diff -Nurd git.orig/arch/arm/tools/mach-types git/arch/arm/tools/mach-types
--- git.orig/arch/arm/tools/mach-types	2014-10-01 12:58:17.354515691 +0200
+++ git/arch/arm/tools/mach-types	2014-10-01 13:01:16.738103183 +0200
@@ -1007,3 +1007,5 @@
 eukrea_cpuimx28sd	MACH_EUKREA_CPUIMX28SD	EUKREA_CPUIMX28SD	4573
 domotab			MACH_DOMOTAB		DOMOTAB			4574
 pfla03			MACH_PFLA03		PFLA03			4575
+himx0280		MACH_HIMX		HIMX			4759
+himx0294		MACH_HIMX0294		HIMX0294		4922
